pulsar-client-cpp/lib/Commands.cc

```diff
@@ -675,7 +675,7 @@ uint64_t Commands::serializeSingleMessageInBatchWithPayload(const Message& msg,
                   << batchPayLoad.writableBytes() << "] can't accomodate new payload [" << requiredSpace
                   << "] - expanding the batchPayload buffer");
         SharedBuffer buffer = SharedBuffer::allocate(batchPayLoad.readableBytes() +
-                                                     std::max(requiredSpace, maxMessageSizeInBytes));
+                                                     std::min(requiredSpace, maxMessageSizeInBytes));
```

Comment:
Why even do minimum then. Internally it's any way a vector so it will reallocate.

Reply:
The `SharedBuffer` class doesn't automatically expand on writes (since it could be just wrapping a memory pointer instead of owning the underlying array). Logically, the `max()` was wrong here, though it could just use `requiredSpace` directly instead because the size check (<maxMessageSize) would have been done before, I guess.